apis/cpi/v1alpha1/cpiconfig_types.go

```diff
@@ -0,0 +1,205 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// CPIConfigSpec defines the desired state of CPIConfig
+type CPIConfigSpec struct {
+	VSphereCPI VSphereCPI `json:"vsphereCPI"`
+}
+
+type NsxTRoute struct {
+	// NSX-T T0/T1 logical router path
+	// +kubebuilder:validation:Optional
+	RouterPath string `json:"routerPath"`
+
+	// Cluster CIDR
+	// +kubebuilder:validation:Optional
+	ClusterCidr string `json:"clusterCidr"`
```

Comment:
This could be a derived field.

Reply:
removed this field, will derived cluster CIDR from the controller